main/tests.py

```diff
@@ -76,4 +78,59 @@ def test_dog_owner_homepage_is_visible_for_dog_owner(self, client, create_dog_ow
         client.force_login(user=create_dog_owner_user.user)
         response = client.get("/homepage/")
         assert response.status_code == 200
-        assert list(response.context['daycares']) == list(DayCare.objects.all())
+        assert set(response.context['day_care_queryset']) == set(DayCare.objects.all())
+
+
+@pytest.mark.django_db
+class TestDogOwnerHomePageView:
+    def test_search_daycare_present_only_available_daycares_on_specific_dates(self, client, create_dog_owner_user):
+        test_day_care = DayCare.create(username='testuser', email='test@email.com', password='test_password',
+                                       name='name', description='test description', price_per_day=100,
+                                       capacity=1, area='CENTER', city='tel aviv', address='address')
+        client.force_login(user=create_dog_owner_user.user)
+        search_form = {'area': "",
+                       'city': "",
+                       'price_per_day': 100,
+                       'name': "",
+                       'start_date': "2022-05-03",
+                       'end_date': "2022-05-08",
+                       }
+        response = client.post('/homepage/', search_form, follow=True)
+        day_care_queryset = response.context['day_care_queryset']
+        assert test_day_care in day_care_queryset
+        Order.create(dog_owner_id=DogOwner.objects.get(id=1), daycare_id=test_day_care,
+                     start_date="2022-05-02", end_date="2022-08-02", price_per_day=100).approve_order()
+        response = client.post('/homepage/', search_form, follow=True)
+        assert test_day_care not in response.context['day_care_queryset']
+
+    def test_successful_dog_owner_search_for_day_care(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        search_form = {'area': "C",
+                       'city': "tel aviv",
+                       'price_per_day': 800,
+                       'name': "",
+                       'start_date': "2022-05-03",
+                       'end_date': "2022-05-08",
+                       }
+        response = client.post('/homepage/', search_form, follow=True)
+        day_care_queryset = response.context['day_care_queryset']
+        available_day_cares = Order.get_all_day_cares_available_on_dates("2022-05-03", "2022-05-08")
+        filters_day_cares = DayCare.objects.filter(area__startswith='C',
+                                                   city__icontains="tel aviv",
+                                                   price_per_day__lte=800)
+        assert set(day_care_queryset) == set(available_day_cares.intersection(filters_day_cares))
```

Comment:
Do we test the functionality of the filter? I mean: do we expect specific search results? what are the day cares that exists during this test?

Reply:
The day-cares that exists during this test are from the migration file . 
In this test I considered the filters queryset as the expected results and make sure that the 'day_care_queryset' from the response is the same .

Reply:
The test should verify the functionality would work in any case, and not only a specific one we carefully created. 

what is `intersection`? what does it provide? This proves the usefulness of declaring the variables before asserting them...

Reply:
NP I will make it more clear .Since we want to filter by capacity of day-care in specific dates we have to create a query set that contain all day-cares avaible in dates , and the second one according to the other filters of the search(we have django built in filter for that ).
The intersection between those querysets will give us all day-cares avaible in specific dates that fit with the other filters .

Do you have an idea how can I make this test more general like you mention ?
Because if I will try all the combination of the filters we won't get out of it. @Omeramsc